### DISCUSSION TITLE
Help Interpreting Sampling Files (AMReX Native Format)

### DISCUSSION THREAD

**Discussion by justhawk98** [Exawind/amr-wind#1305] [Q&A] [answered]
Recently, I implemented a sampling plane in one of my simulations which should be recording velocity and mag_vorticity once a second. AMR-Wind seems to be recording the sampling files correctly, I see them all in post_processing/. But, I'm having trouble interpreting these files. I haven't been able to open them with ParaView like I do plot files and haven't had success with the yt python package either. Does anyone have any scripts they use to visualize these files or otherwise interpret them? Any help is appreciated.

I'm using AMR-Wind v2.1.0, OpenFAST v3.5.2, and the AMR-Wind header says AMReX version 24.03.

Here's an example sampling file:
[sampling01000.tar.gz](https://github.com/user-attachments/files/17466891/sampling01000.tar.gz)

**Comment by marchdf** [comment, marked as answer]
## You should be able to open them with ParaView (point Gaussian display): 
<img width="892" alt="Screenshot 2024-10-21 at 3 18 27 PM" src="https://github.com/user-attachments/assets/620826a8-6df5-4bd4-98ce-ab59289e1272">
<img width="431" alt="Screenshot 2024-10-21 at 3 18 35 PM" src="https://github.com/user-attachments/assets/c273c417-19af-4978-bd26-9c839b6ed107">
<img width="1323" alt="Screenshot 2024-10-21 at 3 19 09 PM" src="https://github.com/user-attachments/assets/e61fb5d1-59af-4c91-a445-edbbeb900c28">

Hopefully these screenshots work. Let me know if it doesn't work.

## Script way:

Here are a couple examples of using python to load the particles:
- https://github.com/Exawind/amr-wind/blob/main/tools/sampling_dam_break_godunov_native.py
- https://github.com/Exawind/amr-wind/blob/main/tools/amrex_particle.py
- https://github.com/Exawind/amr-wind/blob/main/tools/fcompare_particles.py

Let me know if you have issues with those. But basically you get a dataframe of the particles. I've never used `yt` for particles, just fields. So I can't speak to that.

**Reply by justhawk98** [reply]
Thanks for your help! The issue seems to be the version of ParaView I was using (v5.12). I downloaded v5.10.1 which I saw you using and now it works just fine. Strange, I was using the same AMReX boxlib/particles reader in v5.12 and would just get an error when I tried to open the sampling file. Thanks again!

**Reply by marchdf** [reply]
Weird but good to know!
